spec/requests/rent_charges_spec.rb

```diff
@@ -0,0 +1,118 @@
+      expect(rent_charges_gateway.all(2014).first.address_2).to eq(rent_charges.first[:address_2])
+      expect(rent_charges_gateway.all(2014).first.comments).to eq(rent_charges.first[:comments])
+      expect(rent_charges_gateway.all(2014).first.rr_count).to eq(rent_charges.first[:rr_count])
+      expect(rent_charges_gateway.all(2014).first.property_type).to eq(rent_charges.first[:property_type])
+      expect(rent_charges_gateway.all(2014).first.base_data_bed_size).to eq(rent_charges.first[:base_data_bed_size])
+      expect(rent_charges_gateway.all(2014).first.bedroom_weight).to eq(rent_charges.first[:bedroom_weight])
+      expect(rent_charges_gateway.all(2014).first.mra_archetype).to eq(rent_charges.first[:mra_archetype])
+      expect(rent_charges_gateway.all(2014).first.jan_1999_asset_values).to eq(rent_charges.first[:jan_1999_asset_values])
+      expect(rent_charges_gateway.all(2014).first.year).to eq(rent_charges.first[:year] + 1)
+      expect(rent_charges_gateway.all(2014).first.formula_rent_this_year).to eq(168.3)
+      expect(rent_charges_gateway.all(2014).first.rent_cap_this_year).to eq(178.2)
+      expect(rent_charges_gateway.all(2014).first.uprated_actual).to eq(128.7)
+
+      expect(rent_charges_gateway.all(2014).second.uprn).to eq(rent_charges.third[:uprn])
+      expect(rent_charges_gateway.all(2014).second.address).to eq(rent_charges.third[:address])
+      expect(rent_charges_gateway.all(2014).second.address_2).to eq(rent_charges.third[:address_2])
+      expect(rent_charges_gateway.all(2014).second.comments).to eq(rent_charges.third[:comments])
+      expect(rent_charges_gateway.all(2014).second.rr_count).to eq(rent_charges.third[:rr_count])
+      expect(rent_charges_gateway.all(2014).second.property_type).to eq(rent_charges.third[:property_type])
+      expect(rent_charges_gateway.all(2014).second.base_data_bed_size).to eq(rent_charges.third[:base_data_bed_size])
+      expect(rent_charges_gateway.all(2014).second.bedroom_weight).to eq(rent_charges.third[:bedroom_weight])
+      expect(rent_charges_gateway.all(2014).second.mra_archetype).to eq(rent_charges.third[:mra_archetype])
+      expect(rent_charges_gateway.all(2014).second.jan_1999_asset_values).to eq(rent_charges.third[:jan_1999_asset_values])
+      expect(rent_charges_gateway.all(2014).second.year).to eq(rent_charges.first[:year] + 1)
+      expect(rent_charges_gateway.all(2014).second.formula_rent_this_year).to eq(123.75)
+      expect(rent_charges_gateway.all(2014).second.rent_cap_this_year).to eq(138.6)
+      expect(rent_charges_gateway.all(2014).second.uprated_actual).to eq(188.1)
+    end
+
+    it 'saves fixed data row' do
+      expect(fixed_data_gateway.all.first[:year]).to eq(2014)
+      expect(fixed_data_gateway.all.first[:rc_uplift]).to eq(-0.01)
+    end
+
+    it 'returns status code 201' do
+      expect(response).to have_http_status(201)
+    end
+  end
+end
```

Comment:
Happy path tests look good in here. Suggest we add a couple for invalid inputs as well, so we can show basic error handling.

Reply:
Nice changes 👍